experimental/AlgebraicStatistics/src/MultigradedImplicitization.jl

```diff
@@ -0,0 +1,326 @@
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[s[1], s[2]], QQMPolyRingElem[t[1], t[2]])
+
+julia> phi = hom(R, S, [s[1]*t[1], s[1]*t[2], s[2]*t[1], s[2]*t[2]])
+Ring homomorphism
+  from multivariate polynomial ring in 4 variables over QQ
+  to multivariate polynomial ring in 4 variables over QQ
+defined by
+  x[1, 1] -> s[1]*t[1]
+  x[2, 1] -> s[1]*t[2]
+  x[1, 2] -> s[2]*t[1]
+  x[2, 2] -> s[2]*t[2]
+
+julia> max_grading(phi)
+[1   0   0   0   1   1   0   0]
+[0   1   0   0   0   0   1   1]
+[0   0   1   0   1   0   1   0]
+[0   0   0   1   0   1   0   1]
+```
+"""
+function max_grading(phi::MPolyAnyMap)
+  codom = codomain(phi)
+  dom = domain(phi)
+  elim_ring, z = polynomial_ring(QQ, vcat(symbols(codom), symbols(dom)))
+  lift_codom = hom(codom, elim_ring, gens(elim_ring)[1:ngens(codom)])
+  lift_dom = hom(dom, elim_ring, z[ngens(codom) + 1:ngens(elim_ring)])
+
+  return homogeneity_space(ideal([lift_dom(x) - lift_codom(phi(x)) for x in gens(dom)]))
+end
+
+@doc raw"""
+    jacobian(phi::MPolyAnyMap)
+
+Computes the Jacobian of a polynomial map $\phi: \mathbb{K}[x_1, \ldots, x_n] \to \mathbb{K}[t_1, \ldots, t_m]$ such that $(i, j)$ entry is $\frac{d \phi(x_j)}{d t_i}$.
+
+## Examples
+
+```jldoctest
+julia> R, x = polynomial_ring(QQ, :x => (1:2, 1:2))
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[x[1, 1] x[1, 2]; x[2, 1] x[2, 2]])
+
+julia> S, s, t = polynomial_ring(QQ, :s => 1:2, :t => 1:2)
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[s[1], s[2]], QQMPolyRingElem[t[1], t[2]])
+
+julia> phi = hom(R, S, [s[1]*t[1], s[1]*t[2], s[2]*t[1], s[2]*t[2]])
+Ring homomorphism
+  from multivariate polynomial ring in 4 variables over QQ
+  to multivariate polynomial ring in 4 variables over QQ
+defined by
+  x[1, 1] -> s[1]*t[1]
+  x[2, 1] -> s[1]*t[2]
+  x[1, 2] -> s[2]*t[1]
+  x[2, 2] -> s[2]*t[2]
+
+julia> jacobian(phi)
+[t[1]   t[2]      0      0]
+[   0      0   t[1]   t[2]]
+[s[1]      0   s[2]      0]
+[   0   s[1]      0   s[2]]
+```
+"""
+function jacobian(phi::MPolyAnyMap)
+  return matrix(codomain(phi), [[derivative(phi(j), i) for j in gens(domain(phi))] for i in gens(codomain(phi))])
+end
+
+# compute the jacobian and evalauate it at the point pt
+function jacobian_at_point(phi::MPolyAnyMap, pt::fpFieldElem)
+  return matrix(codomain(phi), [[evaluate(derivative(phi(j), i), pt) for j in gens(domain(phi))] for i in gens(codomain(phi))])
+end
+
+# compute the jacobian at a random point with parameters sampled from a finite field
+function jacobian_at_rand_point(phi::MPolyAnyMap; char::UInt=UInt(32003))
+  K = fpField(char)
+  
+  # remake codomain over finite field
+  R = codomain(phi)
+  fq_R, fq_R_gens = polynomial_ring(K, ngens(R))
+
+  # randomly sample parameter values in the finite field
+  pt = rand(K, ngens(fq_R))
+
+  return matrix(K, [[evaluate(derivative(fq_R(phi(j)), i), pt) for j in gens(domain(phi))] for i in fq_R_gens])
```

Comment:
```suggestion
  return map_entries(f->evaluate(map_coefficients(c->K(numerator(c))*inv(K(denominator(c))), f), pt), jacobian(phi))
```
This should work in principle. However
  * We should use `Base.fix1` instead of anonymous functions.
  * We should check that we don't step on a division by zero beforehand (i.e. without `try-catch` constructions.

Reply:
do we know that the entries of the jacobian are always integer? @bkholler

Reply:
left a comment here for now, and we can improve upon later